apps/web-giddh/src/app/vouchers/advance-search/advance-search.component.ts

```
@@ -342,38 +342,49 @@ export class AdvanceSearchComponent implements OnInit, OnDestroy {
+                    // If the field value is a string in the format YYYY-MM-DD, reformat it
+                    fieldValue = dayjs(fieldValue, GIDDH_DATE_FORMAT_YYYY_MM_DD).format(GIDDH_DATE_FORMAT);
+                } else if (typeof fieldValue === 'object') {
+                    fieldValue = dayjs(fieldValue).format(GIDDH_DATE_FORMAT);
+                    if (DATE_REGEX.test(fieldValue)) {
+                        // If the field value is a string in the format YYYY-MM-DD, reformat it
+                        fieldValue = dayjs(fieldValue, GIDDH_DATE_FORMAT_YYYY_MM_DD).format(GIDDH_DATE_FORMAT);
+                    }
+                }
+
+                // Check if the fieldName exists in advanceFilters, and update it
+                if (this.advanceFilters.hasOwnProperty(fieldName)) {
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)